dashboard/app/models/course.rb

```diff
@@ -121,4 +121,57 @@ def summarize
       end
     }
   end
+
+  @@course_cache = nil
+  COURSE_CACHE_KEY = 'course-cache'.freeze
+
+  def self.should_cache?
+    return false if Rails.application.config.levelbuilder_mode
+    ENV['UNIT_TEST'] || ENV['CI']
+  end
+
+  # generates our course_cache from what is in the Rails cache
+  def self.course_cache_from_cache
+    Course.connection
+    # make sure possible loaded objects are completely loaded
+    [CourseScript, Plc::Course].each(&:new)
+    Rails.cache.read COURSE_CACHE_KEY
+  end
+
+  def self.course_cache_from_db
+    {}.tap do |cache|
+      Course.all.pluck(:id).each do |course_id|
+        course = get_without_cache(course_id)
+        cache[course.name] = course
+        cache[course.id.to_s] = course
```

Comment:
Why retrieve all the ids, then loop through and retrieve each Course via `get_without_cache`, instead of one query? 

If this is for the included associations, how about defining them in a scope and using that in both locations?